service/notifications.go

```diff
@@ -0,0 +1,125 @@
+package service
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/RedHatInsights/sources-api-go/config"
+	"github.com/RedHatInsights/sources-api-go/kafka"
+	l "github.com/RedHatInsights/sources-api-go/logger"
+	m "github.com/RedHatInsights/sources-api-go/model"
+)
+
+var NotificationProducer Notifier
+
+type Notifier interface {
+	EmitAvailabilityStatusNotification(accountNumber string, emailNotificationInfo *m.EmailNotificationInfo) error
+}
+
+type AvailabilityStatusNotifier struct {
+}
+
+var (
+	application                = "sources"
+	notificationTopic          = config.Get().KafkaTopic("platform.notifications.ingress")
+	statusEventType            = "availability-status"
+	bundle                     = "console"
+	notificationMessageVersion = "v1.1.0"
+)
+
+type notificationPayload struct {
+}
+
+type notificationEvent struct {
+	Metadata string `json:"metadata"`
+	Payload  string `json:"payload"`
+}
+
+type notificationMetadata struct {
+}
+
+type notificationRecipients struct {
+	OnlyAdmins            bool     `json:"only_admins"`
+	IgnoreUserPreferences bool     `json:"ignore_user_preferences"`
+	Users                 []string `json:"users"`
+}
+
+type notificationMessage struct {
+	Version     string                   `json:"version"`
+	Bundle      string                   `json:"bundle"`
+	Application string                   `json:"application"`
+	EventType   string                   `json:"event_type"`
+	Timestamp   string                   `json:"timestamp"`
+	AccountID   string                   `json:"account_id"`
+	Context     string                   `json:"context"`
+	Events      []notificationEvent      `json:"events"`
+	Recipients  []notificationRecipients `json:"recipients"`
+}
+
+func (producer *AvailabilityStatusNotifier) EmitAvailabilityStatusNotification(accountNumber string, emailNotificationInfo *m.EmailNotificationInfo) error {
+	mgr := &kafka.Manager{Config: kafka.Config{
+		KafkaBrokers:   config.Get().KafkaBrokers,
+		ProducerConfig: kafka.ProducerConfig{Topic: notificationTopic},
+	}}
+
+	context, err := json.Marshal(emailNotificationInfo)
+	if err != nil {
+		l.Log.Warnf("error in marshalling context ")
+		return err
+	}
+
+	payload, err := json.Marshal(&notificationPayload{})
+	if err != nil {
+		l.Log.Warnf("error in marshalling payload ")
+		return err
+	}
+
+	metadata, err := json.Marshal(&notificationMetadata{})
+	if err != nil {
+		l.Log.Warnf("error in marshalling payload ")
+		return err
+	}
```

Comment:
I see that we are marshalling empty structs here. I guess this is related to what Jacob said, right? The payload requires these to be on it.